tests/Feature/Service/LanguageServiceTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Service;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use App\Service\Student\LanguageService;
+use Tests\Fixtures\Students;
+use Tests\Fixtures\Resumes;
+use App\Exceptions\StudentNotFoundException;
+use App\Exceptions\LanguageNotFoundException;
+use App\Exceptions\ResumeNotFoundException;
+
+use Illuminate\Support\Str;
+
+class LanguageServiceTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    protected $languageService;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->languageService = new languageService();
+    }
+
+    public function testLanguageServiceReturnsValidLanguageDetailsForStudentWithLanguages()
+    {
+        $student = Students::aStudent();
+
+        $studentId = $student->id;
+
+        $resume = Resumes::createResumeWithModality($studentId, 'frontend', ['tag1', 'tag2'], 'Presencial');
+        
+        $languages = [];
+
+        for ($i = 0; $i < 2; $i++) {
+            $language = $resume->languages()->create([
+                'language_id' => Str::uuid(),
+                'language_name' => 'Language ' . ($i + 1),
+                'language_level' => 'Bàsic',
+            ]);
+            $languages[] = [
+                'language_id' => $language->id,
+                'language_name' => $language->language_name,
+                'language_level' => $language->language_level,
+            ];
+        }
+
+        $response = $this->languageService->execute($student->id);
+
+        $this->assertIsArray($response);
```

Comment:
I'm I little bit confused with this ammount of asserts. I'll watch them with calm later :sweet

Reply:
The idea is to create a small "battery" of asserts to ensure that, for the proposed test case (returning two languages), both the structure and the quantity of returned elements, as well as their content, are correct.